src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -57,9 +57,18 @@ static PSVersionInfo()
         {
             s_psVersionTable = new PSVersionHashTable(StringComparer.OrdinalIgnoreCase);
 
+            string assemblyPath = typeof(PSVersionInfo).GetTypeInfo().Assembly.Location;
+            FileVersionInfo fileVersionInfo = FileVersionInfo.GetVersionInfo(assemblyPath);
+
+            // Convert to '6.0.0-beta.7-29-g52c6bfe1eae24dbaa1a162bffb3754ba3fdc1f4c'.
+            // The raw git commit version string is widely used internally.
+            string rawGitCommitId = "v" + fileVersionInfo.ProductVersion.Replace(" Commits: ", "-").Replace(" SHA: ", "-g");
```

Comment:
This won't work when it's built from a release tag. When `Commits:` is not present, SHA shouldn't be included in the reconstructed `GitCommitId`.

Reply:
Please clarify. It seems works as expected:
```
> "6.0.0-beta.7".Replace(" Commits: ", "-").Replace(" SHA: ", "-g")
6.0.0-beta.7
```

Reply:
`fileVersionInfo.ProductVersion` will never just be `6.0.0-beta.7`. In case of building with a release tag, it will be like this (for beta.7): 
```
PS> $s.VersionInfo.ProductVersion
6.0.0-beta.7 SHA: d2fe36fb4137c2a7d6dbc0df22245d14256f3a74
```

Reply:
Please clarify. 
1. I see `v6.0.0-beta.7` in Beta.7 distributive  powershell.version file. In Build.psm1 I see the pattern for ReleaseTag "^v\d+\.\d+\.\d+(-\w+\.\d+)?$". If I run PowerShell I see:
```
PowerShell v6.0.0-beta.7
Copyright (C) Microsoft Corporation. All rights reserved.

PS C:\Program Files\PowerShell\6.0.0-beta.7>
```
-- no SHA have we.

2. "6.0.0-beta.7 SHA: d2fe36fb4137c2a7d6dbc0df22245d14256f3a74".Replace(" Commits: ", "-").Replace(" SHA: ", "-g")
Return: v6.0.0-beta.7-gd2fe36fb4137c2a7d6dbc0df22245d14256f3a74
-- the code works as expected.

Reply:
Here is the `$PSVersionTable` for beta.7. Note that `GitCommitId` is ` v6.0.0-beta.7`.
```
PS:1> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      6.0.0-beta
PSEdition                      Core
GitCommitId                    v6.0.0-beta.7
OS                             Microsoft Windows 10.0.15063
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```
The `ProductVersion` of `System.Management.Automation.dll` is
```
PS:6> $s = Get-Item C:\Arena\pscore\System.Management.Automation.dll
PS:7> $s.VersionInfo.ProductVersion
6.0.0-beta.7 SHA: d2fe36fb4137c2a7d6dbc0df22245d14256f3a74
```
And thus based on your code, `GitCommitId` will become:
```
PS> "v" + $s.VersionInfo.ProductVersion.Replace(" Commits: ", "-").Replace(" SHA: ", "-g")
v6.0.0-beta.7-gd2fe36fb4137c2a7d6dbc0df22245d14256f3a74
``` 
and it's obviously not `v6.0.0-beta.7`.

When building from a release tag, the `GitCommitId` should be exactly the same as of today when using `powershell.version` file.

Reply:
😄 I understood that I don't understand.
In official build Beta.7 we have $PSVertionTable.PSVersion= "6.0.0-beta".
In the PR  biuild - "6.0.0-beta.7" - it is hard coded. Should we fix this and remove ".7"?

Reply:
Hope you understand my concern about the `GitCommitId` :)

As for `PSVersion`, it's safer to keep the same as it is, so just change your current code to use `productVersion.LastIndexOf('.')`
```
s_psV6Version = new SemanticVersion(productVersion.Substring(0, productVersion.LastIndexOf('.')));
```
For example, `6.0.0-beta.7 SHA: d2fe36fb4137c2a7d6dbc0df22245d14256f3a74` will get us `6.0.0-beta`

Reply:
Yes,  GitCommitId and  PSVersion mess with my head.

About "point". What will RTM release tag? "v6.0.0-RTM"? If so I'd prefer to split by space and set PSVersion= "6.0.0-beta.7" and PSVersion= "6.0.0-RTM". If we is sensetive to ".7" or "Beta.7" we catch problems with "RTM".

Reply:
I think by RTM, it will be just `v6.0.0`. I'm fine with having `PSVersion= "6.0.0-beta.7"`.
@SteveL-MSFT @JamesWTruher Do any of you have concerns?

Reply:
Yes, the final version should just be `v6.0.0`.  `PSVersion="6.0.0-beta.N"` is correct.  Soon we should have `6.0.0-ReleaseCandidate` and if needed, `6.0.0-ReleaseCandidate.2`

Reply:
I'm not a fan of completely removing the GitCommitId field. I don't mind if it has another name, but it's very useful for our telemetry and understanding how the product is being used and in what way. 

I see the GitCommitId and PSVersion being somewhat related, where GitCommitId is the most specific way to refer to the version of powershell being invoked

Reply:
@daxian-dbw
Changing PSVersion to be a proper representation of any possible semantic version shouldn't break our telemetry. IIRC our telemetry is based on the GitCommitID (@SteveL-MSFT will know for sure). In any event, If it (our telemetry) doesn't support semantic versions, I would argue that we address that, as we _should_ handle a proper semantic version anyway.

Reply:
Thanks @SteveL-MSFT and @JamesWTruher, appreciate your inputs.
@iSazonov Then let's go with `PSVersion= "6.0.0-beta.n`

Reply:
Yes, I believe our telemetry is using GitCommitID and not PSVersion